File: backend/model/src/main/java/courses/paint/mini/model/course/Course.java
package courses.paint.mini.model.course;

import courses.paint.mini.model.User;
import courses.paint.mini.model.game.Miniature;
import lombok.*;

import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = {"id", "title"})
public class Course {

    private String id;
    private String title;
    private Set<CourseStep> steps;
    private Miniature miniature;
    private User user;

}

File: frontend/src/shared/model/course.ts
import {CourseStep} from "./course-step";
import {CourseMiniature} from "./course-miniature";
import {UserDto} from "./user-dto";

export class Course {

  constructor(id: string | null,
              title: string,
              shortDescription: string,
              steps: CourseStep[],
              miniature: CourseMiniature,
              user: UserDto | null) {
    if (id) {
      this.id = id;
    }
    this.title = title;
    this.shortDescription = shortDescription;
    this.steps = steps;
    this.miniature = miniature;
    if (user) {
      this.user = user;
    }
  }

  id!: string;
  title!: string;
  shortDescription!: string;
  steps!: CourseStep[];
  miniature!: CourseMiniature
  user!: UserDto;


}
